components/wpt-results.html

```diff
@@ -0,0 +1,107 @@
+<!--
+Copyright 2017 Google Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+<link rel="import" href="/bower_components/polymer/polymer-element.html">
+<link rel="import" href="/bower_components/polymer/lib/elements/dom-repeat.html">
+
+<dom-module id="wpt-results">
+  <template>
+    <style>
+      :host {
+        display: block;
+      }
+      .WPTResults__spec {
+        background-color: #eee;
+        padding: 1em;
+        margin-bottom: 1em;
+        border: solid 1px #ccc;
+      }
+    </style>
+
+    <template is="dom-repeat" items="{{specs}}">
+      <div class="WPTResults__spec">
+        <b>{{item.spec}}</b><br/>
+        {{item.passing}} / {{item.total}}<br/>
+        {{_passingPercent(item)}}%
+      </div>
+    </template>
+  </template>
+
+  <script>
+    class WPTResults extends Polymer.Element {
+      static get is() { return 'wpt-results' }
+
+      static get properties() {
+        return {
+          specs: {
+            type: Array,
+            observer: '_specsChanged',
+            value: []
+          }
+        };
+      }
+
+      connectedCallback() {
+        super.connectedCallback();
+
+        this._fetchResults()
+      }
+
+      _specFromTestPath(path) {
+        return path.split('/')[1]
+      }
+
+      async _fetchResults() {
+        const url = 'https://storage.googleapis.com/wptdashboard.appspot.com/results/e2d2cde03d/chrome-60.0-debian-8.json.gz'
+        const response = await fetch(url)
+        const data = await response.json()
```

Comment:
> Do we want to use "await" for fetch() and response.json() in order to make this not wait and block the main JS thread?

Yep! Both `fetch()` and `response.json()` return promises, so it's mandatory to handle them asynchronously. `await` just accepts a promise, so these two are functionally equivalent:

async/await
```
const response = await fetch(url)
```
promise
```
fetch(url).then(function (response) {
  ... do something with response
})
```

> (Is there a "main JS thread")?

Yes, JS is single threaded and executes in the rendering thread, so if you do expensive DOM operations you can [freeze the renderer](http://jankfree.org/).

> Are the HTML elements actually displaying the results in the template above also updated then? If so, is that independent of _specsChanged?

Yes, the HTML bindings are updated after `this.specs` is changed since that object is being observed (though via what means I'm not sure). So `_specsChanged` fires independently.

Also, this code [won't work out of the box on Edge](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function#Browser_compatibility), so in a future PR I'll undo the async/await syntax and handle the promises directly.